3p/x-domain-child-message-handler.js

```diff
@@ -0,0 +1,88 @@
+import './polyfills';
+import {listen} from '../src/event-helper';
+import {user} from '../src/log';
+/**
+ * @abstract
+ */
+export class XDomainChildMessageHandler {
+
+  /**
+   *  @param {Window} win A window object.
+   */
+  constructor(win) {
+    /** @private {!Window} */
+    this.win_ = win;
+    /** Map messageType keys to callback functions for when we receive
+     *  that message
+     *  @private {object}
+     */
+    this.callbackFor_ = {};
+    this.sentinel = this.getSentinel();
```

Comment:
should we just move this member field to subclass? Base class should always get it by `this.getSentinel()`